### PR TITLE
Ticket #1194: Disable New Requests

### LINKED ISSUE #1194: For initial launch, remove the ability for users in stable to create new requests
### Issue description

Our November users shouldn't have the ability to request new domain names since that's not opening until later. We should prevent them from starting a new request _in stable_ but it should stay in staging and individual sandboxes so it can continue to be tested.

(The solution here may be a feature flag with e.g., `open_requests=Y/N`.)

### Acceptance criteria

- [ ] Users in stable can't create a new domain request by clicking the "Start a new domain" button, because the button is disabled and greyed out.
- [ ] Users in stable CAN still create a new domain request by using the direct link
- [ ] In stable environment, add language above the "Start a new" button to indicate "Coming in 2024."

(other environments)
- [ ] Users in all other environments can create a new domain request

### Additional context

We could remove this button. We could also keep the button but remove the link behind it (though without updated text it could confuse users).

Screenshot 2023-10-23 at 1 59 21 PM

### Links to other issues

Relates to #1143

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #1194

## Changes
- Added a system flag `IS_PRODUCTION`, added it to settings.py, and passed it into index.py
- Added a tooltip on the `start a new domain request` button with associated styling

## Context for reviewers
This ticket concerns the `Start a new domain request` button on the home page. We want to disable this for MVP, but only on production. I am using a is_production setting to track this. Its currently in my manifest for testing, but I will move it when this PR is ready to merge

## Setup
- [Navigate to my sandbox](https://getgov-za.app.cloud.gov/)
- Try to click the button, bypass it, etc
- Try navigating to this [url](https://getgov-za.app.cloud.gov/register/), you should still be allowed to create new domains requests from here.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [x] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [x] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [x] Pulled this branch locally and tested it
- [x] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [x] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [x] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: ops/manifests/manifest-za.yaml [hunk 1]
```diff
@@ -23,6 +23,9 @@ applications:
     DJANGO_LOG_LEVEL: INFO
     # default public site location
     GETGOV_PUBLIC_SITE_URL: https://beta.get.gov
+    # NOTE: This needs to go in manifest-staging
+    # This is here for PR purposes only!
+    IS_PRODUCTION: True
```
Comment (zandercymatics) [inline ops/manifests/manifest-za.yaml:28]: This will be moved to manifest-stable before merging

**Review by abroddrick — APPROVED** [review]
Love the tool tip! Approving!

File: ops/manifests/manifest-za.yaml (hunk 1 above)
Comment (neilmb) [reply ops/manifests/manifest-za.yaml:28]: I think that it's time to move it now.

**Review by neilmb — COMMENTED** [review]
LGTM (annoying constructive feedback: I'd have loved to see a screenshot in the Pull Request.)

**Comment by gabydisarli** [comment]
I don't see the update on https://getgov-za.app.cloud.gov in safari. Is something wrong there?

**Comment by gabydisarli** [comment]
Can we make the tooltip background D6E9F2 it's named Accent-cool Lightest 15 in Figma

**Comment by gabydisarli** [comment]
Is the tooltip just the out of the box one from USWDS, no adjustment to size? The reason I ask is my component in figma is 36px tall and I'm getting 39.25px on inspect. Not sure. Truss maintains the Figma USWDS file so it could just be our version is off.

**Comment by zandercymatics** [comment]
@gabydisarli The 39.25px value comes from the height of the button, since the tooltip matches its height. Looks like a design mismatch between the original figma vs the developer implementation when they first created the page

**Comment by gabydisarli** [comment]
I'm getting funny behavior on voice over. When I navigate to the button I hear the button link get read and then it reads the below
Screenshot 2023-11-07 at 11 14 06 AM

When I use the shortkeys to click the button, nothing happens, it doesn't say it's disabled. When I press the shortkeys to access "more content" It doesn't read the tool tip text, but I do see a window from VO with the text. If I'm not mistaken, I think the tooltip should read on VO but I can't get it to do so. Can you try it out for yourself and see if you can replicate?

**Comment by gabydisarli** [comment]
(non-blocking) I know this is a uswds component so maybe this is nothing of concern but wanted to flag that the disabled button is not meeting contrast requirements 
Screenshot 2023-11-07 at 11 19 21 AM

**Comment by gabydisarli** [comment]
Actually, maybe we should go with what USWDS has for styling for their disabled buttons. 
Screenshot 2023-11-07 at 11 21 10 AM

The Figma uses some customized buttons that depart from USWDS here, but I think we should go with the latest from USWDS, especially if it's flagging as low contrast

**Comment by zandercymatics** [comment]
@gabydisarli Thanks for taking a look at these! I think our internal styling is being wonky causing some of these issues... Will look into this and get it fixed

**Comment by zandercymatics** [comment]
@gabydisarli Got that done for you
image

**Comment by gabydisarli** [comment]
And just for you info, I think the old disabled styling stems from our Figma USWDS library being out of date with the latest release. Our Figma library is at 3.4.1 and currently USWDS is at 3.5. There is a backlog ticket at Truss to update this Figma library (which I help maintain) so hopefully soon this will be fixed.

**Comment by gabydisarli** [comment]
Great additions. I'm still noticing that VO reads off this as a dimmed link, not a button. Looking at the difference between VO's reading of our button and the [USWDS button](https://designsystem.digital.gov/components/button/#main-content) I see a slightly different order. In USWDS, the label gets read off first, but in ours it's last. Not sure if that is really important or not, but flagging just in case.

our  button
Screenshot 2023-11-07 at 9 31 03 PM

uswds button
Screenshot 2023-11-07 at 9 37 25 PM

**Comment by zandercymatics** [comment]
@gabydisarli This is a consequence of how the page was designed initially. The button is technically an <a> tag rather than a <button>, which I left untouched for the purposes of this ticket. Given that it was this way before, I would personally recommend separating this out into its own ticket. Is this blocking in your eyes?

**Comment by gabydisarli** [comment]
> @gabydisarli This is a consequence of how the page was designed initially. The button is technically an `<a>` tag rather than a `<button>`, which I left untouched for the purposes of this ticket. Given that it was this way before, I would personally recommend separating this out into its own ticket. Is this blocking in your eyes?

Thanks for clarifying. I can make this into a separate ticket. LGTM!

**Review by gabydisarli — APPROVED** [review]

**Pull request merged**